src/data/option/basic.lean

```diff
@@ -384,4 +384,40 @@ def cases_on' : option α → β → (α → β) → β
   cases_on' o (f none) (f ∘ coe) = f o :=
 by cases o; refl
 
+section
+open_locale classical
+
+/-- An arbitrary `some a` with `a : α` if `α` is nonempty, and otherwise `none`. -/
+noncomputable def choice (α : Type*) : option α :=
+if h : nonempty α then
+  some h.some
+else
+  none
+
+lemma choice_eq {α : Type*} [subsingleton α] (a : α) : choice α = some a :=
```

Comment:
Should this be `simp`?

Reply:
Ah, it can't be due to `a`. But perhaps `unique` is better anyway?
```suggestion
@[simp]
lemma choice_eq {α : Type*} [unique α] : choice α = some (default a) :=
```

Reply:
I'll add this as a separate lemma, but actually for the sake of dependent types the first one is actually useful: using `subsingleton` you can rewrite `choice α` as the particular `some a` that you want!

Reply:
Ah, `unique` isn't even defined yet here.